cloud/ec2_price.go

```diff
@@ -24,16 +25,28 @@ type timeRange struct {
 type cachingPriceFetcher struct {
 	ec2Prices map[odInfo]float64
 	ebsPrices map[string]float64
-	m         sync.Mutex
+	sync.Mutex
 }
 
 var pkgCachingPriceFetcher *cachingPriceFetcher
 
-func (cpf *cachingPriceFetcher) getEC2Cost(client AWSClient, h *host.Host, t timeRange) (float64, error) {
-	os := osLinux
-	if strings.Contains(h.Distro.Arch, "windows") {
-		os = osWindows
+func getPkgCachingPriceFetcher() *cachingPriceFetcher {
+	if pkgCachingPriceFetcher == nil {
+		pkgCachingPriceFetcher = new(cachingPriceFetcher)
+	}
+	pkgCachingPriceFetcher.Lock()
+	return pkgCachingPriceFetcher
+}
+
+func putPkgCachingPriceFetcher(fetcher *cachingPriceFetcher) {
+	defer fetcher.Unlock()
+	if fetcher != pkgCachingPriceFetcher {
+		grip.EmergencyPanic("fetcher passed to putPkgCachingPriceFetcher is not package fetcher")
 	}
+}
```

Comment:
I worry that this is going to be pretty unexpected. Let's work on getting an alternate approach to this together.

Reply:
Sure.

Reply:
I pushed a different implementation that doesn't use the get/put indirection.